packages/design-system/src/scalars/components/number-field/number-field.tsx

```diff
@@ -80,30 +121,52 @@ export const NumberFieldRaw = forwardRef<HTMLInputElement, NumberFieldProps>(
             {label}
           </FormLabel>
         )}
-        <Input
-          id={id}
-          name={name}
-          className={cn(
-            // Allow the arrows step
-            step && "show-arrows",
-            className,
+        <div className="relative flex items-center">
+          <Input
+            id={id}
+            name={name}
+            className={className}
+            pattern={isBigInt ? regex.toString() : pattern?.toString()}
+            type={inputType}
+            min={minValue}
+            max={maxValue}
+            aria-valuemin={minValue}
+            aria-valuemax={maxValue}
+            aria-invalid={!!errors?.length}
+            onKeyDown={blockInvalidChar}
+            value={displayValue}
+            defaultValue={defaultValue}
+            onChange={onChange}
+            step={step}
+            onPaste={blockInvalidPaste}
+            ref={ref}
+            {...props}
+          />
+          {step && (
```

Comment:
We should "hide" the steps only when they're `0` IMO. By default they should be enabled and the default value should be `1`. This is how number inputs behave by default in HTML. Take a look to this example: https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_number